### PR TITLE
Defer to Foundation for Path Normalization in Windows

### PR DESCRIPTION
**Pull request by gmittert**
Foundation will do the same thing: removing . and .. and will also have
the benefit of normalizing the windows paths to use '/' instead of '\'

### CODE REVIEW

**Comment by gmittert** [comment]
@swift-ci please smoke test

File: Sources/TSCBasic/Path.swift [hunk 1]
```diff
@@ -561,11 +561,13 @@ private func mayNeedNormalization(absolute string: String) -> Bool {
 ///
 /// The normalization rules are as described for the AbsolutePath struct.
 private func normalize(absolute string: String) -> String {
+  #if os(Windows)
+    return string.standardizingPath
```
Comment (aciidb0mb3r) [inline Sources/TSCBasic/Path.swift]: The precondition and assert are in the `if` branch. That seems wrong?

Comment (gmittert) [reply Sources/TSCBasic/Path.swift]: That does seem wrong... fixed.

**Comment by gmittert** [comment]
@swift-ci please test

**Comment by gmittert** [comment]
@swift-ci please smoke test

**Pull request merged**